lib/Conversion/TorchToMhlo/BasicOp.cpp

```diff
@@ -151,53 +154,44 @@ class ConvertAtenAddSubOp : public OpConversionPattern<AtenOpT> {
     Type outElemTy = outType.getElementType();
     if (!outElemTy.isIntOrFloat()) {
       return op.emitError(
-          "Only floating-point or integer datatype legalization supported");
+          "only floating-point or integer datatype legalization supported");
     }
 
-    Value rhsAsTensor;
     if (!rhsType) {
-      if (failed(mhlo::torchScalarToMhloTensor(rewriter, op, op.other(),
-                                               rhsAsTensor, outElemTy,
-                                               outType.getShape())))
-        return op.emitError("Currently only scalar constants are supported for "
+      if (failed(mhlo::torchScalarToMhloTensor(rewriter, op, op.other(), rhs,
```

Comment:
It seems `torchScalarToMhloTensor` only deal with a constant operand, but the operand#1 could be Value scalar:

``` mlir
func @torch.aten.add.scalar.float(%arg0: !torch.vtensor<[?,?,?,4],f32>, %arg1: !torch.float) -> !torch.vtensor<[?,?,?,4],f32> {
  %int1 = torch.constant.int 1
  %0 = torch.aten.add.Scalar %arg0, %arg1, %int1 : !torch.vtensor<[?,?,?,4],f32>, !torch.float, !torch.int -> !torch.vtensor<[?,?,?,4],f32>
  return %0 : !torch.vtensor<[?,?,?,4],f32>
}
``` 

maybe you can leave a TODO comment here and improve this in the future.